catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -204,6 +212,26 @@ QueryResponse doQuery(QueryRequest queryRequest, FederationStrategy strategy)
 
         LOGGER.debug("Calling strategy.federate()");
 
+        Query originalQuery = queryRequest.getQuery();
+
+        long timeout = originalQuery.getTimeoutMillis();
+
+        if (timeout == 0) {
+            timeout = queryTimeoutMillis;
```

Comment:
Removing infinite timeouts is *way* outside the scope of just adding the current timeout to the queue. So I would push for this to be removed until we have discussion about it. ( We need to make sure that we're fine with removing the ability to do a timeout-less query completely. (@pklinef / @lessarderic))

If we do change it we would need to update the javadoc there to reflect its current behavior and ensure the defaults are still what we want them to be. And if we want to completely disable infinite timeout we would have to make this logic `timeout <= 0`.